### PR TITLE
SDI-138 Add endpoints into csr-api

### PR DESCRIPTION
**Pull request by steverendell**
Added 4 new endpoints (the top 4 in this swagger):

image

### CODE REVIEW

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/ControllerAdvice.kt [hunk 1]
```diff
@@ -29,8 +29,8 @@ class ControllerAdvice {
       .body(ErrorResponse(status = (HttpStatus.INTERNAL_SERVER_ERROR.value()), developerMessage = (e.message)))
   }
 
-  @ExceptionHandler(AccessDeniedException::class)
-  fun handleAccessDeniedException(e: AccessDeniedException?): ResponseEntity<ErrorResponse> {
+  @ExceptionHandler(org.springframework.security.access.AccessDeniedException::class)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/ControllerAdvice.kt:32]: not quite following this change - why do you need to fully qualify?

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt [hunk 2]
```diff
@@ -1,20 +1,28 @@
...
 import org.springframework.web.bind.annotation.RequestParam
 import org.springframework.web.bind.annotation.RestController
 import uk.gov.justice.digital.hmpps.csr.api.dto.DetailDto
 import uk.gov.justice.digital.hmpps.csr.api.service.DetailService
 import java.time.LocalDate
 
-@Tag(name = "details", description = "Get details of shifts from CSR")
+@Tag(
+  name = "details",
+  description = "Details of shifts from CSR. NOTE: All of these endpoints use the header value 'X-Region' to select the required region"
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt:24]: would be better to add a header parameter to the relevant methods
i.e. something like
```
		@Parameter(in = ParameterIn.HEADER, name = "X-Region")

```

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt [hunk 3]
```diff
@@ -29,15 +37,51 @@ class DetailController(private val detailService: DetailService) {
     return ResponseEntity.ok(details)
   }
 
-  @Operation(summary = "Retrieve all modified shifts for a plan unit")
-  @GetMapping("planUnit/{planUnit}/shifts/updated")
+  @Operation(summary = "Retrieve all modified shifts for a plan unit", deprecated = true)
```
Comment (petergphillips) [inline src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt:40]: would expect this to also be deprecated in kotlin with the `@Deprecated`  - might automatically add `deprecated = true` to the endpoint...

**Review by petergphillips — APPROVED** [review]
LGTM

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/ControllerAdvice.kt (hunk 1 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/ControllerAdvice.kt:32]: There was confusion with a kotlin built in AccessDeniedException, but can sort this with an import

File: src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt (hunk 2 above)
Comment (steverendell) [reply src/main/java/uk/gov/justice/digital/hmpps/csr/api/controllers/DetailController.kt:24]: 👍

**Pull request merged**